### PR TITLE
Replace carriage return when getting provider secrets for registry + E2E changes

### PR DESCRIPTION
**Pull request by kaovilai**
[OADP-463](https://issues.redhat.com//browse/OADP-463)
Close https://github.com/openshift/oadp-operator/issues/621

### CODE REVIEW

File: tests/e2e/dpa_deployment_suite_test.go [hunk 1]
```diff
@@ -499,13 +533,23 @@ var _ = Describe("Configuration testing for DPA Custom Resource", func() {
 					if installCase.DpaSpec.BackupLocations[0].Velero.Config != nil {
 						installCase.DpaSpec.BackupLocations[0].Velero.Config["credentialsFile"] = "bsl-cloud-credentials-aws/cloud"
 					}
+					if installCase.TestCarriageReturn {
+						installCase.DpaSpec.BackupLocations[0].Velero.Config["credentialsFile"] = "bsl-cloud-credentials-aws-with-carriage-return/cloud"
```
Comment (kaovilai) [inline tests/e2e/dpa_deployment_suite_test.go:537]: Using carriage return secret

File: tests/e2e/dpa_deployment_suite_test.go [hunk 2]
```diff
@@ -62,6 +63,39 @@ var _ = Describe("Configuration testing for DPA Custom Resource", func() {
 			},
 			WantError: false,
 		}, nil),
+		Entry("Default velero CR, test carriage return", InstallCase{
+			Name:         "default-cr",
+			BRestoreType: RESTIC,
+			TestCarriageReturn: true,
```
Comment (kaovilai) [inline tests/e2e/dpa_deployment_suite_test.go:69]: Added a test for carriage return case

File: tests/e2e/utils/common_utils.go [hunk 3]
```diff
@@ -37,3 +38,9 @@ func WriteFile(credFile string, data []byte) error {
 	err := ioutil.WriteFile(credFile, data, 0644)
 	return err
 }
+
+func ReplaceSecretDataNewLineWithCarriageReturn(data []byte) []byte {
```
Comment (kaovilai) [inline tests/e2e/utils/common_utils.go:42]: New function to replace newline with carriage return in secret data.

File: tests/e2e/e2e_suite_test.go [hunk 4]
```diff
@@ -103,6 +107,8 @@ var _ = BeforeSuite(func() {
 			cloudCreds := GetAzureCreds(cloudCredData)
 			err = CreateCredentialsSecret(cloudCreds, namespace, "bsl-cloud-credentials-azure")
 			Expect(err).NotTo(HaveOccurred())
+			err = CreateCredentialsSecret(utils.ReplaceSecretDataNewLineWithCarriageReturn(cloudCreds), namespace, "bsl-cloud-credentials-azure-with-carriage-return")
```
Comment (kaovilai) [inline tests/e2e/e2e_suite_test.go:110]: Create another secret to use to test carriage return case.

**Review by dymurray — APPROVED** [review]
LGTM, thanks for the tests

File: pkg/credentials/credentials.go [hunk 5]
```diff
@@ -33,7 +33,6 @@ var (
 			IsCloudProvider:    true,
 			SecretName:         "cloud-credentials",
 			MountPath:          "/credentials",
-			BslSecretName:      "bsl-cloud-credentials-aws",
```
Comment (deepakraj1997) [inline pkg/credentials/credentials.go:36]: Kinda followed the logic we had for the previous ones as I included this for the case of CI, now that we are pushing this for new feature, we expect the user to have created the secret name with key given in the `credentialsFile` right ?

Comment (kaovilai) [reply pkg/credentials/credentials.go:36]: I think user should be able to pick which secret is used. My change didn't break functionality so I think it's okay.

Comment (kaovilai) [reply pkg/credentials/credentials.go:36]: Prior to this PR, user was expected to have created a secret anyway but the name was hardcoded. This meant I couldn't use a different secret for the carriage return case unless I overwrite the current secret with the fixed name.

This only affected the credentialFile config case which most customers won't be using. With this PR the secret name for registry is now flexible if they specified bsl secret with credentialsFile config.

Comment (kaovilai) [reply pkg/credentials/credentials.go:36]: >  we expect the user to have created the secret name with key given in the credentialsFile right?

Yes.

**Pull request merged**